File: Junior/FALL_2022/CSE_3302_Programming_Languages/Lab1/nxl9317_lab01.java
// Nghia Lam
// 1001699317
// Windows 
// Using java 16.0.2

import java.io.File;

public class nxl9317_lab01 {
    public static void main(String[] args) 
    {

        File file = new File(".");
        int totalSize = calSize(file.listFiles()); //gets a list of files
        System.out.println(totalSize);
    }

    static int calSize(File[] allFiles)     
    {
        int size = 0;
        for(File fName: allFiles)                   //loop though the files
        {
            if(fName.isDirectory())                 //check if it is a directory
            {
                //System.out.println(fName);
                size += calSize(fName.listFiles()); //does recursion with the directory and adds the size

            }
            if(fName.isFile())                      //check if it is a file
            {
                //System.out.println(fName);
                size += fName.length();
                //System.out.println(size);
            }

        }
        return size;
    }

}

File: Junior/FALL_2022/CSE_3302_Programming_Languages/Lab1/nxl9317_lab01.py
# Nghia Lam
# 1001699317
# Windows 
# Using python 3.9.7

import os

location = '.'			#This is the current working directory

def sizeCal(ldir):		
	size = 0 			

	for file in os.scandir(ldir):		#loops for every file in the directory

		if file.is_file():				#if it is a file
			file_stats = os.stat(file)	#gets the stats of the file then put it in size
			size += file_stats.st_size
			#print(size)

		if file.is_dir():				#if it is a directory
			#print(file.path)
			size += sizeCal(file)		#recursively does the function and puts it into size

	return size

total_size = sizeCal(location)
print(total_size)
